src/uwtools/rocoto.py

```diff
@@ -31,10 +31,10 @@ def realize_rocoto_xml(
     """
     rxml = _RocotoXML(config)
     xml = str(rxml).strip()
-    if not validate_rocoto_xml_string(xml):
-        raise UWError("Internal error: Invalid Rocoto XML")
     with writable(output_file) as f:
         print(xml, file=f)
+    if not validate_rocoto_xml_string(xml):
+        raise UWError("Internal error: Invalid Rocoto XML")
```

Comment:
Adding this here gives a tangible but incorrect file that is easier for debugging.

Reply:
Hmm, thats a fair point, I think the `.bad`/`,err`/etc would be good, or we could dump to `stdout` in the conditional too

Reply:
Only output consumable by downstream tools (including `uw` via pipes) should be written to `stdout`. Our logging is configured to write to `stderr`, so iterating over the lines and logging them would be fine as an alternative to writing to a file with an extension.

Reply:
I will take this change out for now and may revisit it later when the time crunch is reduced a bit.